packages/webpack/src/plugin.ts

```diff
@@ -58,19 +73,81 @@ export class NgcWebpackPlugin {
     this.genDir = this.options.genDir
                || path.resolve(process.cwd(), this.angularCompilerOptions.genDir + '/app');
     this.entryModule = options.entryModule || (this.angularCompilerOptions as any).entryModule;
+    if (!options.entryModule && options.main) {
+      this.entryModule = this.figureOutEntryFromMain(options.main);
+    }
 
     const entryModule = this.entryModule;
     const [rootModule, rootNgModule] = entryModule.split('#');
     this.projectPath = options.project;
     this.rootModule = rootModule;
     this.rootModuleName = rootNgModule;
-    this.compilerHost = ts.createCompilerHost(this.compilerOptions, true);
+    this.compilerHost = new WebpackCompilerHost(this.compilerOptions);
     this.program = ts.createProgram(this.files, this.compilerOptions, this.compilerHost);
     this.reflectorHost = new ngCompiler.ReflectorHost(
       this.program, this.compilerHost, this.angularCompilerOptions);
     this.reflector = new ngCompiler.StaticReflector(this.reflectorHost);
   }
 
+  figureOutEntryFromMain(mainPath: string): string {
+    const source = ts.createSourceFile(
+      mainPath, fs.readFileSync(mainPath, 'utf-8'), ts.ScriptTarget.Latest);
+
+    const bootstrap = _findNodes(source, source, ts.SyntaxKind.CallExpression, false)
+      .map(node => node as ts.CallExpression)
+      .filter(call => {
+        const access = call.expression as ts.PropertyAccessExpression;
+        return access.kind == ts.SyntaxKind.PropertyAccessExpression
+            && access.name.kind == ts.SyntaxKind.Identifier
+            && (access.name.text == 'bootstrapModule'
+             || access.name.text == 'bootstrapModuleFactory');
+      });
+
+    if (bootstrap.length != 1
```

Comment:
It doesn't support it. Having multiple bootstraps in a single file is just bad. Please note that in this case, the CLI doesn't support it but you can specify which specific module you want to bootstrap with `entryModule` in your webpack config.